### PR TITLE
Add launch_socket_server

### PR DESCRIPTION
**Pull request by jamer**
### New Formulae Submissions:

- [⨉] Does your submission pass
`brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [✓] Have you built your formula locally prior to submission with `brew install <formula>`?

### Issues

The project has 48 Github stars but `brew audit` wants 50 so it gives an error.

### CODE REVIEW

File: Library/Formula/launch_socket_server.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    (libexec/"launch_socket_server").install "libexec/launch_socket_server/login_wrapper"
+  end
+
+  def caveats; <<-EOS.undent
+    launch_socket_server is designed to be run under launchd and must be
+    configured before running.
+
+    See https://github.com/sstephenson/launch_socket_server for details.
+    EOS
+  end
```
Comment (bfontaine) [inline Library/Formula/launch_socket_server.rb]: You can remove these as they’re not Homebrew-specific.

Comment (jamer) [inline Library/Formula/launch_socket_server.rb]: Removed!

File: Library/Formula/launch_socket_server.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>KeepAlive</key>
+        <true/>
+        <key>ProgramArguments</key>
+        <array>
+          <string>#{opt_sbin}/launch_socket_server</string>
+          <string>/usr/local/bin/myserver</string>
```
Comment (mikemcquaid) [inline Library/Formula/launch_socket_server.rb]: This isn't an actual binary so you can delete the the whole `plist` here.

Comment (jamer) [inline Library/Formula/launch_socket_server.rb]: Okay, removed.

File: Library/Formula/launch_socket_server.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  def install
+    system "make"
+
+    sbin.install "sbin/launch_socket_server"
+    (libexec/"launch_socket_server").install "libexec/launch_socket_server/login_wrapper"
+  end
+
+  test do
+    assert_match /usage/, shell_output("#{sbin}/launch_socket_server")
```
Comment (mikemcquaid) [inline Library/Formula/launch_socket_server.rb]: Is it possible to run a command e.g. `true` or `/usr/bin/env` with this to be able to test the actual functionality beyond just usage instructions? Thanks!

Comment (jamer) [inline Library/Formula/launch_socket_server.rb]: Thanks for the prompt! 🙂 Wrote a new test for the actual functionality.

File: Library/Formula/launch_socket_server.rb [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    system "launchctl", "load", testpath/"launch_socket_server.plist"
+    system "sleep", "1"
+
+    s = TCPSocket.new "127.0.0.1", launch_port
+    s.puts "hello, world"
+    output = s.gets.strip
+    s.close
+
+    system "launchctl", "unload", testpath/"launch_socket_server.plist"
```
Comment (mikemcquaid) [inline Library/Formula/launch_socket_server.rb]: Worth putting this in an `begin`/`ensure` block.

File: Library/Formula/launch_socket_server.rb [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    (testpath/"echo_server.rb").write <<-EOS.undent
+      require "socket"
+
+      server = TCPServer.new("127.0.0.1", "#{echo_port}")
+      socket = server.accept
+      socket.puts socket.readline
+    EOS
+
+    system "launchctl", "load", testpath/"launch_socket_server.plist"
+    system "sleep", "1"
```
Comment (mikemcquaid) [inline Library/Formula/launch_socket_server.rb]: Can just use `sleep 1` to use the Ruby version.

Comment (jamer) [inline Library/Formula/launch_socket_server.rb]: Changed to use this.

File: Library/Formula/launch_socket_server.rb (hunk 4 above)
Comment (jamer) [inline Library/Formula/launch_socket_server.rb]: I've wrapped the TCPSocket with the ensure block, but did not include the the plist load. I'm assuming that if the load fails then we don't need to unload.

**Comment by apjanke** [comment]
Could you squash this to a single commit and rebase it on `master` so we have a tidy commit history? That'll trigger a rebuild to get fresh bottle files, too.

**Comment by apjanke** [comment]
@BrewTestBot test this please

**Comment by apjanke** [comment]
Build failure is just `audit` complaining about the star count. I'm proceeding to pull it, and just squashing the commits as part of the pull.

Merged. Thank you for your contribution to Homebrew!

**Comment by apjanke** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/4570b0a6a163ccc851d22d453c9c6661dc459798

**Pull request closed without merging**